Mathlib/Data/Finset/Image.lean

```diff
@@ -657,6 +660,56 @@ theorem biUnion_singleton {f : α → β} : (s.biUnion fun a => {f a}) = s.image
 
 end Image
 
+/-! ### filterMap -/
+
+section FilterMap
+
+/-- `filterMap f s` is a combination filter/map operation on `s`.
+  The function `f : α → Option β` is applied to each element of `s`;
+  if `f a` is `some b` then `b` is included in the result, otherwise
+  `a` is excluded from the resulting finset.
+
+  In notation, `filterMap f s` is the finset `{b : β | ∃ a, f a = some b}`. -/
+-- TODO: should there be `filterImage` too?
+def filterMap (f : α → Option β) (s : Finset α)
+    (f_inj : ∀ a a' b, b ∈ f a → b ∈ f a' → a = a') : Finset β :=
+  ⟨s.val.filterMap f, s.nodup.filterMap f f_inj⟩
+
+@[simp]
+theorem filterMap_val (f : α → Option β) (s : Finset α)
+    {f_inj : ∀ a a' b, b ∈ f a → b ∈ f a' → a = a'} :
```

Comment:
```suggestion
theorem filterMap_val (f : α → Option β) (s : Finset α) {f_inj} :
```

Reply:
I moved the `variable` declaration up instead